content/projects/2022-implicit/index.md

```diff
@@ -1,10 +1,103 @@
+
+## Implicit Backpropagation 
+Suppose now we have found a fixed point $x^\ast$ for a single layer. Then, $$ x^\ast = T_\Theta (x^\ast) $$
+
+Using implicit differentiation on the equation above we have, 
+
+$$\frac{d x^\ast}{d \Theta} = \left( I - \frac{d T_\Theta (x^\ast)}{d x^\ast}\right)^{-1} \frac{\partial T_\Theta (x^\ast)}{\partial \Theta}$$ 
+
+However, solving this is very expensive because of the inverse term.
+
+To circumvent this issue, we use a recently proposed method called Jacobian-Free Backpropagation.
+
+## Jacobian-Free Backpropagation (JFB)
+The goal of JFB is to alleviate memory requirement and avoid high computational cost in implicit networks. 
+
+The key idea is to replace the problematic Jacobian $$\left( I - \frac{d T_\Theta (x^\ast)}{d x^\ast}\right)$$ with the identity matrix $I$. 
+
+For a comparison, if we were to calculate the true gradient using implicit networks we have the following equation: 
+
+$$\nabla_\Theta \ell = \frac{d \ell}{d x^\ast} \left( I - \frac{d T_\Theta (x^\ast)}{d x^\ast}\right) ^{-1} \frac{\partial T_\Theta (x^\ast)}{\partial \Theta}$$
+
+Using JFB to approximate the gradient we only need to solve: 
+$$p_\Theta = \frac{d \ell}{d x^\ast} \frac{\partial T_\Theta (x^\ast)}{\partial \Theta}$$ 
+which is a descent direction for the loss $\ell$. 
+
+Utilizing JFB, we avoid computing the Jacobian term. As a result, implicit networks are trained faster and more easily implemented.  
+
+Note: the JFB approach relies on a set of conditions to be true:
+- $T_\Theta$ is contraction mapping with Lipschitz constant $\gamma$
+- $T_\Theta$ is continuously differentiable w.r.t. $\Theta$
+- $M := \frac{\partial T_\Theta}{\partial \Theta}$ has full column rank 
+- $M$ is well-conditioned, i.e., $\kappa (M^T M) < \frac{1}{\gamma}$
+
+## Numerical Experiments
+In our project we used the CelebA dataset, which consist of annotated celebrity faces. The images are categorized into various sections based on specific features that the celebrities have. For example, whether or not they have bangs, wear glasses, have a pointy nose, etc. 
+![CelebA](imgs/celebA.jpg)
+
+## Results
+The results are as follows
+![loss plot](imgs/loss_plot-1.jpg)
+From the graph we see that the loss is decreasing as the number of epochs increases. An epoch is one complete pass of the entire dataset through our algorithm. 
+![results](imgs/truth_36-1.jpg)
+Note: Two metrics are commonly used for assessing the quality of reconstructed images: the peak-signal-to-noise ratio (PSNR, a positive number, best at $+\infty$) and the structural similarity index measure (SSIM, also positive, best at $1$).
+
+## Acknowledgements
+We sincerely thank the guidance of our mentor, Dr. Samy Wu Fung, and other mentors at Emory University for the opportunity. This work is supported in part by the US National Science Foundation awards DMS-2051019 and DMS-1751636.
```

Comment:
No need to list the grants here. They are listed on our main website.  OK of course to leave the first sentence if you like.